test/integration/M3/Base.t.sol

```diff
@@ -0,0 +1,219 @@
+pragma solidity ^0.8.24;
+
+import {TransparentUpgradeableProxy, ITransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+
+import {IStrategy} from "@eigenlayer-contracts/interfaces/IStrategyManager.sol";
+import {IEigenPodManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IEigenPodManager.sol";
+import {IDelegationManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+import {BeaconChainMock, BeaconChainProofs, CheckpointProofs, CredentialProofs, EigenPodManager} from "lib/eigenlayer-contracts/src/test/integration/mocks/BeaconChainMock.t.sol";
+
+import {Utils} from "../../../script/Utils.sol";
+import {ContractAddresses} from "../../../script/ContractAddresses.sol";
+import {ActorAddresses} from "../../../script/Actors.sol";
+
+import {IDepositContract} from "../../../src/external/ethereum/IDepositContract.sol";
+
+import {IRedeemableAsset} from "../../../src/interfaces/IRedeemableAsset.sol";
+import {IRedemptionAssetsVault} from "../../../src/interfaces/IRedemptionAssetsVault.sol";
+import {IynETH} from "../../../src/interfaces/IynETH.sol";
+
+import {ynETH} from "../../../src/ynETH.sol";
+import {StakingNodesManager, IStakingNodesManager} from "../../../src/StakingNodesManager.sol";
+import {StakingNode} from "../../../src/StakingNode.sol";
+import {RewardsReceiver} from "../../../src/RewardsReceiver.sol";
+import {RewardsDistributor} from "../../../src/RewardsDistributor.sol";
+import {StakingNode} from "../../../src/StakingNode.sol";
+import {WithdrawalQueueManager} from "../../../src/WithdrawalQueueManager.sol";
+import {ynETHRedemptionAssetsVault} from "../../../src/ynETHRedemptionAssetsVault.sol";
+
+import "forge-std/console.sol";
+import "forge-std/Test.sol";
+
+contract Base is Test, Utils {
+
+    // Utils
+    ContractAddresses public contractAddresses;
+    ContractAddresses.ChainAddresses public chainAddresses;
+    ActorAddresses public actorAddresses;
+    ActorAddresses.Actors public actors;
+
+    // Rewards
+    RewardsReceiver public executionLayerReceiver;
+    RewardsReceiver public consensusLayerReceiver;
+    RewardsDistributor public rewardsDistributor;
+
+    // Staking
+    StakingNodesManager public stakingNodesManager;
+    StakingNode public stakingNodeImplementation;
+
+    // Assets
+    ynETH public yneth;
+
+    // Withdrawals
+    WithdrawalQueueManager public ynETHWithdrawalQueueManager;
+    ynETHRedemptionAssetsVault public ynETHRedemptionAssetsVaultInstance;
+
+    // EigenLayer
+    IEigenPodManager public eigenPodManager;
+    IDelegationManager public delegationManager;
+
+    // Mock Contracts to deploy
+    BeaconChainMock public beaconChain;
+
+    // Ethereum
+    IDepositContract public depositContractEth2;
+
+    address public constant GLOBAL_ADMIN = 0x72fdBD51085bDa5eEEd3b55D1a46E2e92f0837a5;
```

Comment:
this GLOBAL_ADMIN should NOT be used at all here.

It was a temporary measure for the preprod, we must not use it here in any way shape or form

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/48f72390ec79e2ad439f89d86f244eb6ebc0c7e5